### PR TITLE
Bump docs infraestructure

### PR DESCRIPTION
**Pull request by echedey-ls**
- ~[ ] Closes #xxxx~
 - ~[ ] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)~
 - ~[ ] Tests added~
 - ~[ ] Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - ~[ ] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.~
 - [ ] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

I hate going so much behind the latest versions. Here is the docs updated.

# Issues
- View on GitHub text icon is a pencil, not the previous GH logo
  - Extending the template now only allows modifying the text, not the logo
    - See https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/source-buttons.html#custom-link-text
    - which overrides https://github.com/pydata/pydata-sphinx-theme/blob/main/src/pydata_sphinx_theme/theme/pydata_sphinx_theme/components/edit-this-page.html
- Pages without any section still feature the "Section Navigation" title on the left side bar, see [example](https://pvlib-python--2112.org.readthedocs.build/en/2112/contributing.html)
  - According to the [Sphinx Pydata Theme](https://pydata-sphinx-theme.readthedocs.io/en/stable/user_guide/layout.html#primary-sidebar-left) this should happen: "When there are no pages to show, it will disappear and potentially add extra space for your page’s content."

# Link
https://pvlib-python--2112.org.readthedocs.build/en/2112/

### CODE REVIEW

**Comment by echedey-ls** [comment]
Ready for review!!
@kandersolar @AdamRJensen @cwhanse @IoannisSifnaios @RDaxini

**Comment by kandersolar** [comment]
@echedey-ls it looks like docstring references are getting two sets of brackets:

image

We have custom CSS to add brackets, but I guess that is no longer needed.  It would be nice if the references were the same height as normal text though (not superscript).  Here is the code: https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/_static/reference_format.css

**Review by cwhanse — COMMENTED** [review]
My review wouldn't add any value here.

**Comment by echedey-ls** [comment]
>  It would be nice if the references were the same height as normal text though (not superscript).

Done @kandersolar !

**Comment by IoannisSifnaios** [comment]
I noticed that the "Next" arrow in the bottom right in the API reference does not always show correctly (I guess it is due to the long function name). To be fair, it was like this in the old site as well, but it might be nice if it is possible to fix it...

image

**Comment by IoannisSifnaios** [comment]
I vote for the first!

**Comment by echedey-ls** [comment]
Another issue with long names:
https://pvlib-python--2112.org.readthedocs.build/en/2112/reference/generated/pvlib.pvsystem.SingleAxisTrackerMount.get_orientation.html

https://pvlib-python.readthedocs.io/en/stable/reference/generated/pvlib.pvsystem.SingleAxisTrackerMount.get_orientation.html

I would not take action on these unless we find a way to break words on the dot `.`

**Comment by kandersolar** [comment]
It would be good to check the `pydata-sphinx-theme` docs and github (if you haven't already) to see if there is some better solution.  We should be wary of relying on CSS too much.  As we've already seen in this PR, CSS tweaks often break when we upgrade to newer versions of the theme.  

An aside:  do people actually like the "previous/next" links?  They have always seemed like clutter to me... I just use the sidebar for all navigation.

**Comment by AdamRJensen** [comment]
> An aside:  do people actually like the "previous/next" links?  They have always seemed like clutter to me... I just use the sidebar for all navigation. 

For stuff like API documentation it seems like clutter.  I vote to nix them

**Comment by echedey-ls** [comment]
>  do people actually like the "previous/next" links? 

I do both in the API and the user guides! When I'm using new features (for me), so I can discover other kind of functions that may do a similar work as the one I'm looking at. I suppose the index is the best place for that anyway.

I think it makes sense to preserve that in the user guide pages, so users can flow with the topics order without browsing the index tree.

Not a strong opinion, one more nix-it and we can remove that IMO.

**Comment by RDaxini** [comment]
> do people actually like the "previous/next" links?

Personally, I've never used them... I also use the sidebar. Chances of someone needing to browse through the functions in the order of those links seems small.

-----
Not sure if it is an issue or just the way the new docs are, but I miss the colour of block code. I feel like the more distinct colours in the old docs (including the In / Out) improves the readability of the code, at least for me anyway.
**New:**
new
**Old:**
old

Surprisingly, I find the slightly smaller text easier to read too. Not sure why... a bit less in-your-face maybe?

**Comment by RDaxini** [comment]
The RMS value on the final figure for the _Fast simulation using the ADR efficiency model starting from PVsyst parameters_ gallery example in [the new docs](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/adr-pvarray/plot_simulate_fast.html) is different to the RMS value in [the old docs ](https://pvlib-python.readthedocs.io/en/stable/gallery/adr-pvarray/plot_simulate_fast.html) (0.16 vs 0.15)
Is this an issue?

**Comment by cwhanse** [comment]
> The RMS value on the final figure for the _Fast simulation using the ADR efficiency model starting from PVsyst parameters_ gallery example in [the new docs](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/adr-pvarray/plot_simulate_fast.html) is different to the RMS value in [the old docs ](https://pvlib-python.readthedocs.io/en/stable/gallery/adr-pvarray/plot_simulate_fast.html) (0.16 vs 0.15) Is this an issue?

@adriesse

**Comment by AdamRJensen** [comment]
I've clicked around a bunch and most things are looking good to me. Thanks for taking this initiative @echedey-ls!

When we're ready, I suggest that we wait and merge this right after the next release. Then all reviews of new PRs will be using the new documentation layout, which gives us time until the next release to passively detect any potential issues.

Note, I've added an issue in the PR description concerning "Pages without any section still feature the "Section Navigation"". Also, it seems to me that the copyright isn't rendered anywhere.

File: readthedocs.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ version: 2
 build:
    os: ubuntu-20.04
    tools:
-     python: "3.8"
+     python: "3.9"
```
Comment (AdamRJensen) [inline readthedocs.yml:11]:
```suggestion
     python: "3.12"
```

**Comment by AdamRJensen** [comment]
Let's do a more rigorous walk-through where we make sure that all pages have been skimmed. Here's a proposal for how to delegate the documentation:

- [ ] Users guide - @RDaxini
- [ ] Example gallery - @IoannisSifnaios
- [ ] API reference (classes, solar position, Clear sky, airmass): @kandersolar 
- [ ] API reference (irradiance, PV modeling, effects on PV System output, tracking): @echedey-ls 
- [ ] Everything else: @AdamRJensen

**Comment by IoannisSifnaios** [comment]
So I skimmed through the entire Example Gallery and found the following issues:

- [ ] All the `View on GitHub` links are broken (this was recently fixed, but maybe the new sphinx uses different paths?)
- [ ] For some reason, the references in examples [1](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/agrivoltaics/plot_diffuse_PAR_Spitters_relationship.html#sphx-glr-gallery-agrivoltaics-plot-diffuse-par-spitters-relationship-py), [2](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/irradiance-transposition/plot_rtranpose_limitations.html#sphx-glr-gallery-irradiance-transposition-plot-rtranpose-limitations-py), [3](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/reflections/plot_diffuse_aoi_correction.html#sphx-glr-gallery-reflections-plot-diffuse-aoi-correction-py), [4](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/shading/plot_martinez_shade_loss.html#sphx-glr-gallery-shading-plot-martinez-shade-loss-py), [5](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/soiling/plot_fig3A_hsu_soiling_example.html#sphx-glr-gallery-soiling-plot-fig3a-hsu-soiling-example-py) render using the old format (where the number of times each citation is used shows up new to them): 
image
However, there were many cases that the references rendered differently. Not sure why this happens...

- [ ] Is it possible to remove these useless `Out:` boxes (unless we actively request something to be printed): 
image

Otherwise, things look fine! :-)

**Comment by echedey-ls** [comment]
>     * [ ]  All the `View on GitHub` links are broken (this was recently fixed, but maybe the new sphinx uses different paths?)

That was me not updating the branch from main 😬 

>     * [ ]  For some reason, the references in examples [...]
> However, there were many cases in which the references were rendered differently. Not sure why this happens...

What is differently? `[1]` just with the link, no parenthesis with the occurrences like `[1](1,2)`? I think that's the default; I'm -1 to CSS-remove it since that would give us some references with the occurrence link and some others without it.

>     * [ ]  Is it possible to remove these useless `Out:` boxes (unless we actively request something to be printed):
> 
> 
> image

You may find this link informative: https://sphinx-gallery.github.io/stable/auto_examples/plot_3_capture_repr.html#matplotlib-output
It seems a sphinx-gallery feature more than a customization left to the end users.

> Otherwise, things look fine! :-)

Thx :) ❤️

**Comment by adriesse** [comment]
> > The RMS value on the final figure for the _Fast simulation using the ADR efficiency model starting from PVsyst parameters_ gallery example in [the new docs](https://pvlib-python--2112.org.readthedocs.build/en/2112/gallery/adr-pvarray/plot_simulate_fast.html) is different to the RMS value in [the old docs ](https://pvlib-python.readthedocs.io/en/stable/gallery/adr-pvarray/plot_simulate_fast.html) (0.16 vs 0.15) Is this an issue?
> 
> @adriesse

Always curious why things change, but I do not see it as a problem.

**Comment by IoannisSifnaios** [comment]
> What is differently? `[1]` just with the link, no parenthesis with the occurrences like `[1](1,2)`? I think that's the default; I'm -1 to CSS-remove it since that would give us some references with the occurrence link and some others without it.

You are correct, my bad. It is just that in some cases, the References looked like the one below, and I thought they had changed something, but I guess the reason was that it was mentioned only once in the text...
image

> You may find this link informative: https://sphinx-gallery.github.io/stable/auto_examples/plot_3_capture_repr.html#matplotlib-output
It seems a sphinx-gallery feature more than a customization left to the end users.

Thank you for the link! Yes, ok, I guess a fix could be to have the `plt.show()` at the end; but this could be for another PR!

**Comment by RDaxini** [comment]
I read through all sections of the user guide and found one issue and one difference.
One issue is that the bullet points and sub-bullet points have disappeared from the User Guide main page in the new docs.
One difference is the layout of the footer. It is not a major issue, but from a stylistic perspective my preference leans towards the old layout.
I will have another scan through at another time just in case I missed anything but this is all I have found for now.
Images for both pointers below:
**Here's the new main page:**
new main
**Here's the old main page:**
old main
**Here's the new footer layout:**
new layout
**Here's the old footer layout:**
old layout

**Comment by AdamRJensen** [comment]
> I read through all sections of the user guide and found one issue and one difference. One issue is that the bullet points and...

@RDaxini You have keen eyes! However, I don't think that's per se an issue but simply a style change. To reduce the maintenance and because the changes are minor, I suggest we simply just accept these small changes.

@echedey-ls in #1359 we made some changes to the position of the placement of the ethical ads (I don't think this fix is necessary any longer). This means that we can leave `html_sidebars` unspecified which defaults to `"**": ["sidebar-nav-bs", "sidebar-ethical-ads"]` and the `footer_end` can be set to empty `{}` if we don't want to show the theme version. `footer_start` and `footer_center` should also then be unspecified, i.e., left to their default values.

**Comment by kandersolar** [comment]
I checked my sections (https://github.com/pvlib/pvlib-python/pull/2112#issuecomment-2210837549) and found nothing that should be fixed in this PR!

**Comment by echedey-ls** [comment]
@AdamRJensen thx for the input on the footer and navbar.

Out of the 140 functions in my sections, I have compared roughly a third, taking special care to features like tables, figures and constants documentation. Everything looks good to me. Maybe the pvsystem class table that lists it is much more prominent in the new docs, thou I don't find it as an issue since it also comes from a well-formed toc tree.

- New: https://pvlib-python--2112.org.readthedocs.build/en/2112/reference/pv_modeling/index.html
- Stable: https://pvlib-python.readthedocs.io/en/stable/reference/pv_modeling/index.html

File: docs/sphinx/source/conf.py [hunk 2]
```diff
@@ -164,12 +162,23 @@
     ],
     "use_edit_page_button": True,
     "show_toc_level": 1,
-    "footer_items": ["copyright", "sphinx-version", "sidebar-ethical-ads"],
-    "left_sidebar_end": [],
-}
+    # "footer_start": [],  # "copyright", "sphinx-version"
+    # "footer_center": [],
+    "footer_end": [],
+    "primary_sidebar_end": [],
```
Comment (AdamRJensen) [inline docs/sphinx/source/conf.py:168]:
```suggestion
    # "primary_sidebar_end": [],
```
This defaults to ``"primary_sidebar_end": ["sidebar-ethical-ads"]``, and should be commented out for ads to show up. Probably an oversight on my behalf.

Comment (echedey-ls) [reply docs/sphinx/source/conf.py:168]: Don't worry, to be fair I also didn't bother enough to double check your suggestion.

**Comment by echedey-ls** [comment]
Looks like https://github.com/pvlib/pvlib-python/pull/2112#issuecomment-2210837549 list is already full. Do you all miss anything else? I think we are good to go, feedback is welcome.

**Comment by kandersolar** [comment]
Everyone has checked their sections, @echedey-ls is happy, let's merge it!  

Thanks @echedey-ls for the initiative, and the broader group for the collective review!

**Review by kandersolar — APPROVED** [review]

**Pull request merged**